raiden/network/transport/matrix/utils.py

```diff
@@ -637,35 +637,40 @@ def validate_userid_signature(user: User) -> Optional[Address]:
     return address
 
 
-def sort_servers_closest(servers: Sequence[str]) -> Dict[str, float]:
+def sort_servers_closest(servers: Sequence[str], max_timeout: float = 2.0) -> Dict[str, float]:
     """Sorts a list of servers by http round-trip time
 
     Params:
         servers: sequence of http server urls
     Returns:
-        sequence of pairs of url,rtt in seconds, sorted by rtt, excluding failed servers
-        (possibly empty)
+        sequence of pairs of url,rtt in seconds, sorted by rtt, excluding failed and excessively
+        slow servers (possibly empty)
     """
     if not {urlparse(url).scheme for url in servers}.issubset({"http", "https"}):
         raise TransportError("Invalid server urls")
 
-    timeout = 1.0
     rtt_greenlets = set(
-        gevent.spawn(return_after_retries, server_url, timeout) for server_url in servers
+        gevent.spawn(get_average_http_response_time, server_url, max_timeout)
+        for server_url in servers
     )
 
-    result = gevent.wait(rtt_greenlets, count=1)[0].get()
+    results = []
+    # `get_average_http_response_time` performs 3 requests with 0.125 s delay inbetween
+    for greenlet in gevent.iwait(rtt_greenlets, timeout=max_timeout * 3 + 1):
```

Comment:
This will iterate over all servers, the result will that server selection will be as fast as the slowest server.

Reply:
Please, don't have multiple timeouts for the same thing ... this is just hard to debug and compose. The timeout here is not necessary since the `get_average_http_response_time` has a timeout already, and it will return `None` if the request itself timedout.